src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -891,19 +893,122 @@ public void testLookUpFieldList() throws IOException, CommandException
+        assertEquals("lookUpField target table should be present", "lookUpList", lookupColumn.getAllProperties().get("lookupQuery"));
+        assertNull("lookUpField target table should be null for current container", lookupColumn.getAllProperties().get("lookupContainer"));
+    }
+
+    @Test
+    public void verifyExpectedWarningOnNavigateWithUncomittedChanges() throws Exception
+    {
+        goToProjectHome();
+        String homeUrl = getDriver().getCurrentUrl();
+        String listName = "dirtyListNavigationTest";
+
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "lists", listName);
+        TestDataGenerator dgen1 = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("favoriteIceCream", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("favoriteSnack", FieldDefinition.ColumnType.String)));
+        DomainResponse createResponse = dgen1.createDomain(createDefaultConnection(true), "IntList", Map.of("keyName", "id"));
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "lists", listName);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldProperties(listName);
+
+        // edit an existing field
+        domainFormPanel.getField("favoriteSnack")
+                .setDescription("Exactly what it sounds like. Someone's fave frozen dairy goodness")
+                .collapse();
+        // create a new field
+        domainFormPanel.addField("newField")
+                .setDescription("Exactly what it sounds like. A new dang field");
+
+        // capture the current url; ensure we don't navigate anywhere when we refresh or browse
+        String currentUrl = getDriver().getCurrentUrl();
+
+        // now refresh view, without saving
+        getDriver().navigate().refresh();
+        // dismiss the alert
+        shortWait().until(ExpectedConditions.alertIsPresent());     // currently alert does not appear in FF; this is a known issue
+        getDriver().switchTo().alert().dismiss();
+
+        // make sure we are still here
+        assertEquals(currentUrl, getDriver().getCurrentUrl());
+
+        // now navigate away
+        getDriver().navigate().to(homeUrl);
+        // dismiss the alert
+        shortWait().until(ExpectedConditions.alertIsPresent());
+        getDriver().switchTo().alert().dismiss();
+        // ensure current location
+        assertEquals(currentUrl, getDriver().getCurrentUrl());
+
+        domainDesignerPage = new DomainDesignerPage(getDriver()); // we've been refreshing; things are possibly stale
+        domainDesignerPage.clickCancel().saveChanges();
+    }
+
+    @Test
+    public void verifyWarningForMarkingExistingSampleSetFieldRequired() throws Exception
```

Comment:
Is this special sample set behavior? If we're just using sample sets to verify universal behavior, having `SampleSet` in the name is misleading.

Reply:
I'm not sure; my understanding was that this needed to work specifically on a sampleset.  It's definitely worth following up to see if this is/should be universal/true for every domain kind